backend/pkg/rest/server.go

```diff
@@ -149,13 +163,21 @@ func CreateRESTServer(port int, speculator *_speculator.Speculator, dbHandler *d
 
 	server.ConfigureFlags()
 	server.ConfigureAPI()
-	server.Port = port
+	server.Port = config.Port
+
+	// We want to serve both http and https
+	if config.EnableTLS {
+		server.EnabledListeners = []string{"https", "http"}
+		server.TLSCertificate = flags.Filename(config.TLSServerCertFilePath)
```

Comment:
maybe we want to check that the certs actually exists?

Reply:
If files are empty or does not exist if should fail during server creation 
```
			httpsServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair(string(s.TLSCertificate), string(s.TLSCertificateKey))

```